src/atlas/interpolation/element/Quad2D.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * (C) Crown Copyright 2021 Met Office
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include <iosfwd>
+#include <limits>
+
+#include "atlas/interpolation/Vector2D.h"
+#include "atlas/interpolation/method/Intersect.h"
+#include "atlas/runtime/Exception.h"
+#include "atlas/util/Point.h"
+
+namespace atlas {
+namespace interpolation {
+namespace method {
+struct Ray;
+}
+namespace element {
+
+#define BAD_WEIGHT_VALUE -1
```

Comment:
```suggestion
#define BAD_WEIGHT_VALUE -1
```
static constexpr double BAD_WEIGHT_VALUE = -1.;